library/Fission/Web/User/Create.hs

```diff
@@ -37,7 +37,8 @@ server
      )
   => ServerT API m
 server (User.Registration username password email) = do
-  void . runDBNow <| User.create username password email Nothing
+  createResponse <- runDBNow <| User.create username password email Nothing
+  void . Web.Err.ensure <| createResponse
```

Comment:
Can save a character if we treat `<|` as `$`.

```haskell
void <| Web.Err.ensure createResponse
```

Any thoughts? This is pretty un-Elixir in convention, but we need to figure out pipe conventions in Haskell as a team IMO

Reply:
I think haskell has a few to many operators so Im all for using a pipe instead since its more widely used.

I think dot syntax if your creating a new function for general use

```haskell
funcZ = funcX . funcY
```

Reply:
Re-reading your response, I think you may have misunderstood what I wrote:

```haskell
void . Web.Err.ensure <| createResponse
void <| Web.Err.ensure createResponse
```

It's not about eliminating the `.`. It's about not needing to delineate every word on a line 😉

Still going to put my initial response below, because some still applies. It'll be clear that I misread what you meant _here_ about "too many operators" 😅

---

# Pipe Count

Interesting — that's certainly an alchemist's view on pipes 😉 I don't mean that in a disparaging way; just genuinely interesting to see the influence.

Clearly, I'm still reading `<|` as `$`. In idiomatic Haskell, this reads like nonsense...

```haskell
foo $ bar
foo <| bar
```

...because `$` is "put everything on the right in brackets"...

```haskell
foo (bar)
```

...which written that way looks absurd.

`$` can also creates nice lines of delineation of values vs functions, something that we're losing here (or at least trading off).

The pipe has _such a strong metaphor_ that I can see the argument. It's a bit noisier, but seems to be the way that people would think about the pipe.

## More Widely Used

> Im all for using a pipe instead since its more widely used.

Is it, though? Right-pipes are nearer at hand for you, but that's very much the influence of Elixir. They have different conventions in Elm and the OCaml/Reason/F# family, where they're generally used as I presented above.

Given that they're totally foreign to Haskell, we get to decide on how we want to use them, but they're neither common to most programmers nor standard in use across languages.

# Operators

## Too Many

> I think haskell has a few to many operators

Sure, but you also haven't written much idiomatic Haskell. `.` and `$` are _by far_ the most common operators in the language, so the argument about there being too many doesn't totally apply. Remember that JS has about 2 dozen, and they're overloaded to do all kinds of stuff. You just don't see them as operators because you learned with them in place. Yes, Haskell can suffer from operator-itus, but certainly not here where I'm asking if you'd like to remove one.

Our code base has:

* `|>` application
* `.` composition
* `\` lambda
* `=` assignment
* `<$>` map
* `<|>` alternate
* `::` type
* `^.` accessor
* `.~` set
* `?~` set maybe
* Inequalities (`==` and the like)

TS has
* `()` application (usually)
* `.` application
* `.` accessor by string key
* `=>` lambda
* `[]` accessor by non-string key
* `{}` closure (usually)
* `=` assignment
* `:` type 
* Inequalities (`==` and the like)

Yes, in Haskell you can define operators, but mostly they're the same ones reused. Most languages have a lot of different kinds of syntax, but they're all in the C-family, so they feel familiar and seem to "disappear" into the langauge.

## Point Free Style

Writing functions without explicit arguments is called the point free style:

```haskell
funcZ = funcX . funcY
```

Sometimes it's very useful, and it's something that I desperately miss when writing Elixir (they're so close!) There are a lot of cases where this is useful, but I'm slowly moving towards the part of the community that says that explicit arguments are actually better on named functions...

```haskell
funcZ arg = arg |> funcY |> funcX
```

...and anonymous functions can be phrased in a nice way by reducing noise...

```haskell
[1,2,3]
  |> fmap (\x -> upcase (show x))
  |> -- ...

[1,2,3]
  |> fmap (upcase . show)
  |> -- ...
```

# Prime

This is all good stuff adding to thinking about what should go into Prime 🎉

Reply:
> > Im all for using a pipe instead since its more widely used.
> 
> Is it, though? Right-pipes are nearer at hand for you, but that's very much the influence of Elixir. 
Oh I had meant in our project specifically!

Reply:
> but I'm slowly moving towards the part of the community that says that explicit arguments are actually better on named functions

I really agree on this point!